recipes/picobench/all/conanfile.py

```diff
@@ -18,54 +21,43 @@ class PicobenchConan(ConanFile):
     default_options = {
         "with_cli": False,
     }
-    generators = "cmake"
+    generators = "CMakeToolchain"
+    exports_sources = ["CMakeList.txt", "patches/*"]
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["PICOBENCH_BUILD_TOOLS"] = self.options.with_cli
+        tc.generate()
 
-    def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 11)
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, 11)
 
     def package_id(self):
-        if self.options.with_cli:
-            del self.info.settings.compiler
-        else:
-            self.info.header_only()
+        self.info.clear()
```

Comment:
```suggestion
        if self.options.with_cli:
            del self.info.settings.compiler
        else:
            self.info.clear()
```
with_cli affects package's content.